botocore/credentials.py

```diff
@@ -122,6 +124,46 @@ def create_credential_resolver(session, cache=None):
     return resolver
 
 
+class ProfileProviderBuilder(object):
```

Comment:
It might be worth noting that this is an internal class and not to use externally.